src/server/scripts/Northrend/Ulduar/Ulduar/boss_assembly_of_iron.cpp

```diff
@@ -986,18 +916,24 @@ class spell_assembly_rune_of_summoning : public SpellScriptLoader
 
             void OnPeriodic(AuraEffect const* aurEff)
             {
-                if (Unit* caster = GetCaster())
-                    if (aurEff->GetTickNumber() % 2 == 0)
-                        caster->SummonCreature(CREATURE_LIGHTNING_ELEMENTAL, caster->GetPositionX(), caster->GetPositionY(), caster->GetPositionZ(), 0, TEMPSUMMON_CORPSE_TIMED_DESPAWN, 2000);
+                PreventDefaultAction();
+                GetTarget()->CastSpell(GetTarget(), SPELL_RUNE_OF_SUMMONING_SUMMON, true, nullptr, aurEff, GetTarget()->IsSummon() ? GetTarget()->ToTempSummon()->GetSummonerGUID() : 0);
```

Comment:
`if (aurEff->GetTickNumber() % 2 == 0)`

Reply:
Add the “if (aurEff->GetTickNumber() % 2 == 0)”
Otherwise the frequency is too high